### PR TITLE
Split out wallet

### PR DESCRIPTION
**Pull request by lalexgap**
Split out the wallet so it can be hosted in its own iFrame.

### CODE REVIEW

File: packages/wallet-comm/src/interface/to-wallet.ts [hunk 1]
```diff
@@ -60,14 +67,13 @@ export const closeChannelRequest = () => ({
 export type CloseChannelRequest = ReturnType<typeof closeChannelRequest>;
 
 
-// VALIDATION
+// VAßLIDATION
```
Comment (kerzhner) [inline packages/wallet-comm/src/interface/to-wallet.ts]:
```suggestion
// VALIDATION
```

File: packages/wallet-comm/src/wallet-functions.ts [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+  iFrame.style.display = 'none';
+  iFrame.style.position = 'absolute';
+  iFrame.style.left = '0px';
+  iFrame.style.right = '0px';
+  iFrame.style.bottom = '0px';
+  iFrame.style.top = '0px';
+  iFrame.width = '0';
+  iFrame.height = '0';
+  iFrame.style.zIndex = '9999';
+  iFrame
```
Comment (kerzhner) [inline packages/wallet-comm/src/wallet-functions.ts]: Is ts linting enabled for this code?

File: packages/wallet-comm/src/wallet-functions.ts [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+}
+/**
+ *  Validates signed data.
+ * Promise resolves when the data is verified or an error occurs.
+ */
+export async function validateSignature(iFrameId: string, data, signature: string): Promise<boolean> {
+  const iFrame = <HTMLIFrameElement>document.getElementById(iFrameId);
+  const message = validationRequest(data, signature);
+  iFrame.contentWindow.postMessage(message, "*");
```
Comment (kerzhner) [inline packages/wallet-comm/src/wallet-functions.ts]: Can there exist a race condition where the response event is fired before the listener is added?

File: packages/wallet-comm/src/wallet-functions.ts (hunk 2 above)
Comment (lalexgap) [reply packages/wallet-comm/src/wallet-functions.ts]: Not yet, that's still a TODO (This one particular case has been cleaned up)

File: packages/wallet-comm/src/wallet-functions.ts (hunk 3 above)
Comment (lalexgap) [reply packages/wallet-comm/src/wallet-functions.ts]: That's a good point, we should probably create the promise before posting the message

**Pull request merged**